spec/success_response_spec.rb

```diff
@@ -6,21 +6,25 @@
 RSpec.describe SuccessResponse do
   let(:weather_body) { File.open('fixtures/london_weather.json').read }
 
-  describe '#body' do
-    context 'if response has an empty body' do
-      it 'creates an error message' do
-        custom_response = described_class.new(successful_response(nil))
+  describe '#ok?' do
+    it 'returns a truthy value if JSON parse is successful' do
+      custom_response = described_class.new(successful_response)
 
-        custom_response.body
+      expect(custom_response.ok?).to be true
+    end
 
-        expect(custom_response.error_message).to include("I couldn't manage to get any weather info.")
-      end
+    it 'returns a falsely value if JSON parse fails' do
+      custom_response = described_class.new(successful_response('irelivent'))
 
-      it 'returns an empty hash' do
-        custom_response = described_class.new(successful_response(nil))
+      expect(custom_response.ok?).to be false
+    end
+  end
 
-        expect(custom_response.body).to eq({})
-      end
+  describe '#body' do
+    it 'returns an empty hash if response has an empty body' do
+      custom_response = described_class.new(successful_response(nil))
```

Comment:
You will never get a `nil` as a response that travels through the Internet, because you can only send and receive strings. You even wrote `empty body` in the test description.
So what you really want is:

```ruby
custom_response = described_class.new(successful_response(''))
```